vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get/get.go

```diff
@@ -583,6 +583,12 @@ func (o *GetOptions) watch(f cmdutil.Factory, cmd *cobra.Command, args []string)
 				return false, nil
 			}
 
+			// if we are watching a single resource, ignore events meant for other resources
+			info, err := meta.Accessor(e.Object)
```

Comment:
You're loosing that `err` if it is non-nil, the below check only get a nil-error.

Reply:
Shouldn't your new condition be connected with the one above, I think it's touching the same area, isn't it?